pkg/kiln/docker_repo.go

```diff
@@ -329,20 +330,23 @@ func (imageCreator LocalImageCreator) PushImage(dockerInfo *DockerInfo) (chan (s
 	return streamParser.Channel(), nil
 }
 
-//GenerateRepoURI generate the repo uri
-func (imageCreator LocalImageCreator) GenerateRepoURI(dockerInfo *DockerInfo) string {
-
-	return dockerInfo.GetRemoteTagName(imageCreator.remoteRepo)
-}
-
 //generateAuthConfiguration Create an auth configuration from the environment variables
 func generateAuthConfiguration(remoteRepo string) string {
+	var authConfig *types.AuthConfig
+	var exists bool
 
-	authConfig, exists := getAuthConfig(remoteRepo)
+	// we will use GCR when deployed on GCP, so we will need to access Google Service Account token
+	if os.Getenv("DOCKER_PROVIDER") == "gcr" {
```

Comment:
Yeah not ideal, you create a GetAuthConfig interface function/struct that could be injected when doing the switch on the registry provider. But it works though not sure if it's the right to be optimizing structure. Who knows...